src/lib/sidebar-components/Feedback/FeedbackModal.tsx

```diff
@@ -0,0 +1,239 @@
+import { useEffect, useState } from "react"
+import CloseIcon from "./CloseIcon"
+import { feedbackCategories, FeedbackContent } from "./Feedback"
+import FeedbackCategoryButton from "./FeedbackCategoryButton"
+import Icon from "./FeedbackIcon"
+import SuccessIcon from "./SuccessIcon"
+
+type Props = {
+  onSubmitFunction: (feedbackPayload: FeedbackContent) => Promise<boolean>
+  closePopup: () => void
+}
+
+function FeedbackModal({ onSubmitFunction, closePopup, ...rest }: Props) {
+  const [feedbackContent, setFeedbackContent] = useState<FeedbackContent>(
+    () => {
+      const storedFeedbackContent = sessionStorage.getItem("feedbackContent")
+      return storedFeedbackContent !== null
+        ? (JSON.parse(storedFeedbackContent) as FeedbackContent)
+        : {
+            emailContent: "",
+            feedbackCategory: undefined,
+            responseRequested: false,
+          }
+    }
+  )
+  const [isLoading, setIsLoading] = useState(false)
+  const [responseMessage, setResponseMessage] = useState<
```

Comment:
@joobacca Irgendwie sieht die Formatierung hier komisch aus

Reply:
Die Formatierung ist schon korrekt, man könnte eventuell die Type definition des useState-hooks in einen type auslagern, da das aber sonst nirgendwo benutzt wird, sollte es in Ordnung sein.